packages/runtime/runtime-utils/src/summaryUtils.ts

```diff
@@ -128,6 +129,10 @@ export class SummaryTreeBuilder implements ISummaryTreeWithStats {
         this.summaryStats = mergeStats(this.summaryStats, summarizeResult.stats);
     }
 
+    public addAttachment(key: string, id: string) {
```

Comment:
I'd remove key argument - there seems to be not much value. SummaryBuilder can use counter to genera keys 1,2,3,4...